tests/capture/test_base_interpreter.py

```diff
@@ -0,0 +1,439 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule tests strategy structure for defining custom plxpr interpreters
+"""
+# pylint: disable=protected-access, too-few-public-methods
+import pytest
+
+import pennylane as qml
+
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+from pennylane.capture import PlxprInterpreter  # pylint: disable=wrong-import-position
+from pennylane.capture.primitives import (  # pylint: disable=wrong-import-position
+    adjoint_transform_prim,
+    cond_prim,
+    ctrl_transform_prim,
+    for_loop_prim,
+    qnode_prim,
+    while_loop_prim,
+)
+
+pytestmark = pytest.mark.jax
+
+
+@pytest.fixture(autouse=True)
+def enable_disable_plxpr():
+    """Enable and disable the PennyLane JAX capture context manager."""
+    qml.capture.enable()
+    yield
+    qml.capture.disable()
+
+
+class SimplifyInterpreter(PlxprInterpreter):
+
+    def interpret_operation(self, op):
+        new_op = op.simplify()
+        if new_op is op:
+            new_op = new_op._unflatten(*op._flatten())
+            # if new op isn't queued, need to requeue op.
+        return new_op
+
+
+# pylint: disable=use-implicit-booleaness-not-comparison
+def test_env_and_initialized():
+    """Test that env is initialized at the start."""
+
+    interpreter = SimplifyInterpreter()
+    assert interpreter._env == {}
+    assert interpreter._op_math_cache == {}
+
+
+def test_primitive_registrations():
+    """Test that child primitive registrations dict's are not copied and do
+    not effect PlxprInterpreeter."""
+
+    class SimplifyInterpreterLocal(PlxprInterpreter):
+
+        def interpret_operation(self, op):
+            new_op = op.simplify()
+            if new_op is op:
+                # if new op isn't queued, need to requeue op.
+                new_op = new_op._unflatten(*op._flatten())
+            return new_op
+
+    assert (
+        SimplifyInterpreterLocal._primitive_registrations
+        is not PlxprInterpreter._primitive_registrations
+    )
+
+    @SimplifyInterpreterLocal.register_primitive(qml.X._primitive)
+    def _(self, *invals, **params):  # pylint: disable=unused-argument
+        print("in custom interpreter")
+        return qml.Z(*invals)
+
+    assert qml.X._primitive in SimplifyInterpreterLocal._primitive_registrations
+    assert qml.X._primitive not in PlxprInterpreter._primitive_registrations
+
+    @SimplifyInterpreterLocal()
+    def f():
+        qml.X(0)
+        qml.Y(5)
+
+    jaxpr = jax.make_jaxpr(f)()
+
+    with qml.queuing.AnnotatedQueue() as q:
+        jax.core.eval_jaxpr(jaxpr.jaxpr, [])
+
+    qml.assert_equal(q.queue[0], qml.Z(0))  # turned into a Y
+    qml.assert_equal(q.queue[1], qml.Y(5))  # mapped wire
```

Comment:
There was no wire mapped, was there? :thinking: 
```suggestion
    qml.assert_equal(q.queue[0], qml.Z(0))  # turned into a Z
    qml.assert_equal(q.queue[1], qml.Y(5))
```